### PR TITLE
Add Beta Datastore Handling

### PR DESCRIPTION
**Pull request by kanikaul-amazon**
## Description
* Handling for Beta Datastore Prefix
* Add WCF Template files as beta

## Additional context
Please provide any additional information related to this PR

---
*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: src/CTA.Rules.Config/Constants.cs [hunk 1]
```diff
@@ -113,6 +115,10 @@ public class Constants
             new List<string> {"mvc","Program.cs"},
             new List<string> {"mvc","Startup.cs" },
             new List<string> {"webclasslibrary","appsettings.json" },
+        };
+
+        public static readonly List<List<string>> BetaTemplateFiles = new List<List<string>>()
```
Comment (marknfawaz) [inline src/CTA.Rules.Config/Constants.cs:120]: Are these different from the other file names? Not sure why we need these..

**Review by marknfawaz — CHANGES_REQUESTED** [review]

File: src/CTA.Rules.PortCore/SolutionPort.cs [hunk 2]
```diff
@@ -177,8 +177,18 @@ internal void DownloadRecommendationFiles(HashSet<string> allReferences)
...
+                        {
+                            // Check Beta Recommendations
+                            var fileContents = httpClient.GetStringAsync(string.Concat(Constants.S3BetaRecommendationsBucketUrl, "/", fileName)).Result;
+                            File.WriteAllText(fullFileName, fileContents);
+                        }
+
+                        catch (Exception)
+                        {
+                            //We are checking which files have a recommendation, some of them won't
+                            SkipDownloadFiles.TryAdd(fullFileName, false);
```
Comment (marknfawaz) [inline src/CTA.Rules.PortCore/SolutionPort.cs:190]: Why are we adding it only in an error case? Do we need to redownload it every time?

**Review by marknfawaz — CHANGES_REQUESTED** [review]

File: src/CTA.Rules.PortCore/SolutionPort.cs (hunk 2 above)
Comment (kanikaul-amazon) [reply src/CTA.Rules.PortCore/SolutionPort.cs]: Moved try catch logic inside the File Download, that way we can check on File Exists

**Review by marknfawaz — APPROVED** [review]
Beta version only

**Pull request merged**